app/(app)/transactions/[id]/page.tsx

```diff
@@ -18,12 +19,18 @@ export default async function TransactionDetailPage({
    * - `!transaction` is true when the transaction is undefined.
    * - `transaction.length === 0` is true when the transaction array is empty.
    */
-  if (error || !transaction || transaction.length === 0) return notFound();
+  if (error || !data || data.length === 0) return notFound();
```

Comment:
This condition is duplicate, remove it or refactor it. You have already handled this condition in the `getTransactionById()` function.

Reply:
Hmmm... I see...